### PR TITLE
Update documentation for logging, analysis, and scheduler

### PR DESCRIPTION
**Pull request by semihokur**

### CODE REVIEW

File: Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ The most powerful setting to modify is the maximum number of child process pips
                                 Specifies maxProc in terms of a multiplier of the machine's processor count. The
                                 default is 1.25.
 
-**Note** - In WDG builds, db.exe automatically sets /maxProcMultiplier to 1.5. So that's the effective default if you are in wdg
+
+**Note** - In WDG builds, db.exe automatically sets /maxProcMultiplier to 1.5. So that's the effective default if you are in WDG.
```
Comment (mpysson) [inline Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md]: no longer relevant

**Comment by mpysson** [comment]
> * HashSourceFile pips

These aren't run anymore, correct? Shall we remove this from the documentation?

---
Refers to: Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md:26 in 540600c. [](commit_id = 540600caa12e1160a44eb8132fd8a7a6cb6c4fbb, deletion_comment = False)

**Comment by mpysson** [comment]
> ## IO Concurrency limit 

Can we add what the default is?

---
Refers to: Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md:24 in 540600c. [](commit_id = 540600caa12e1160a44eb8132fd8a7a6cb6c4fbb, deletion_comment = False)

**Comment by mpysson** [comment]
> By default the server process will exit when no build has been performed for 60 minutes. This value is configurable with `/serverMaxIdleTimeInMinutes `

extra space

---
Refers to: Documentation/Wiki/Advanced-Features/Server-Mode.md:16 in 540600c. [](commit_id = 540600caa12e1160a44eb8132fd8a7a6cb6c4fbb, deletion_comment = False)

File: Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md [hunk 2]
```diff
@@ -26,21 +27,33 @@ Some pips are primitives for IO operations. These are:
...
+Since cache operations use different resources (IO, CPU, network) concurrently, they have a different concurrency limit settings.
 
-    /lowPriority[+|-]           Runs the build engine and all tools at a lower priority in order to provide better
-                                responsiveness to interactive processes on the current machine.
+    /maxCacheLookup:<number of concurrent operations>
+                                Specifies the maximum number of cache lookup operations that BuildXL will launch at one time. 
+                                The default value is 2 times the number of processors in the current machine.
+    /maxMaterialize:<number of concurrent operations>   
+                                Specifies the maximum number of concurrent materialize operations (e.g., materialize inputs, storing two-phase cache entries, analyzing pip violations). 
+                                The default value is 2 times the number of processors in the current machine.
```
Comment (mpysson) [inline Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md]: This section is duplicated

File: Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md [hunk 3]
```diff
@@ -26,21 +27,33 @@ Some pips are primitives for IO operations. These are:
...
 
 ## Memory utilization ##
```
Comment (mpysson) [inline Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md]: > Memory utilization  [](start = 2, length = 20)

Is it worth adding a note that the default values are very conservative which in essence make this feature disabled by default?

File: Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md [hunk 4]
```diff
@@ -26,21 +27,33 @@ Some pips are primitives for IO operations. These are:
...
 BuildXL supports throttling spawned process pips based on machine available RAM. This aims at ensuring builds don't page memory onto disk for machines with low memory to CPU core ratios.
```
Comment (mpysson) [inline Documentation/Wiki/How-To-Run-BuildXL/Resource-Usage-Configuration.md]: >throttling [](start = 17, length = 10)

Mention that this is historic runtime based

File: Documentation/Wiki/How-To-Run-BuildXL/Log-Files/BuildXL.log.md [hunk 5]
```diff
@@ -17,39 +17,52 @@ Factors limiting concurrency by build time: CPU:72%, Graph shape:0%, Disk:2%, Co
...
+```
+[0:16.488] verbose DX5017: Critical path:
+Pip Duration(ms) | Exe Duration(ms)| Queue Duration(ms) | Pip Result   | Scheduled Time | Completed Time | Pip
+            4849 |            2325 |                182 |              |                |                | *Total
+               2 |               0 |                  0 |     Executed |       00:00:15 |       00:00:15 | Pip7998CDAE90476A13 ...
+            3840 |            2325 |                  3 |     Executed |       00:00:12 |       00:00:15 | Pip1FC84BBAC24D0F71 ...
+              11 |               0 |                  0 |       Cached |       00:00:12 |       00:00:12 | Pip4814212D6E796253 ...
 ```
 ### Performance Summary
 This section contains a performance summary of the BuildXL run broken down by phases.
```
Comment (mpysson) [inline Documentation/Wiki/How-To-Run-BuildXL/Log-Files/BuildXL.log.md]: > phases.  [](start = 77, length = 9)

Currently only available for single machine builds

File: Documentation/Wiki/How-To-Run-BuildXL/Log-Files/BuildXL.xlg.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    * Pip data extracted from BuildXL spec files
+    * Build graph
+    * String table and path table with all strings/paths referenced by the build
+
+## How to generate BuildXL Execution Logs
+There are several BuildXL command line arguments that control what is captured in the execution log and your build configuration may have one or more of these defaulted on.
+
+BuildXL command line argument | Description
+--- | ---
+/logExecution | Instructs BuildXL to generate an execution log with the build.
```
Comment (mpysson) [inline Documentation/Wiki/How-To-Run-BuildXL/Log-Files/BuildXL.xlg.md]: >build. [](start = 72, length = 6)

Enabled by default

File: Documentation/Wiki/How-To-Run-BuildXL/Log-Files.md [hunk 7]
```diff
@@ -1,7 +1,13 @@
...
+* BuildXL.status.csv - BuildXL regularly logs build status information to this CSV file every specified seconds (default: 2) during the build. Some of the information that this file contains: Resource (cpu, mem, disks, network) usage, number of pips in each pip state and execution step, worker usage statistics (acquired slots, status, etc.), scheduler statistics. 
+* BuildXL.CacheMiss.log - This file contains the detailed reasons of the cache misses if [Runtime Cache Miss Analysis](../Advanced-Features/Cache-Miss-Analysis.md) feature is enabled.
+* BuildXL.Dev.log - This file is a small subset of [BuildXL.log](./Log-Files/BuildXL.log.md). As the main log can be sometimes very large, developers skim over *Dev.log* to troubleshoot issues. 
+* BuildXL.DistributionRpc.log - This file contains the communication logs between master and workers if distributed build is enabled.
+* BuildXL.PipOutput.log - This file contains the process pip output messages.
```
[the remark this answers is not in the document]
Comment (semihokur) [reply Documentation/Wiki/How-To-Run-BuildXL/Log-Files.md]: yes, it is there for any CloudBuild build.

**Pull request merged**